src/trix/models/block.coffee

```diff
@@ -137,13 +143,13 @@ class Trix.Block extends Trix.Object
     @attributes[depth]
 
   canBeGroupedWith: (otherBlock, depth) ->
-    attributes = @attributes
-    otherAttributes = otherBlock.getAttributes()
-    if attributes[depth] is otherAttributes[depth]
-      if attributes[depth] in ["bullet", "number"] and otherAttributes[depth + 1] not in ["bulletList", "numberList"]
-        false
-      else
-        true
+    otherAttributes = otherBlock.getAttributes() 
+    otherAttribute = otherAttributes[depth]
+    attribute = @attributes[depth]
+
+    attribute is otherAttribute and
+      not (Trix.config.blockAttributes[attribute].groups is false and
+           otherAttributes[depth + 1] not in ["bulletList", "numberList"])
```

Comment:
Let's get the hard coded references to `bulletList` and `numberList` out of here too.